Xamarin.Forms.Platform.Android/AppCompat/FormsAppCompatActivity.cs

```diff
@@ -265,9 +265,16 @@ protected override void OnResume()
 				CurrentFocus.ShowKeyboard();
 			}
 
+			var isRestart = _previousState == AndroidApplicationLifecycleState.OnRestart;
+
 			_previousState = _currentState;
 			_currentState = AndroidApplicationLifecycleState.OnResume;
 
+			if (isRestart)
+			{
+				_application?.SendResume();
+			}
+
```

Comment:
Can we move this logic into `OnStateChanged`? I'm not entirely sure why we didn't sprinkle the logic in `OnStateChanged` into the overrides like you have but since we've already established the precedent let's keep it (or recast call the logic as you have here). For one thing, it'll ensure the lollipop code below runs before the callback is placed.

Reply:
@jsuarezruiz I will have some time this weekend and see what can be done.

Reply:
It's done.